hls.js/index.d.ts

```diff
@@ -0,0 +1,523 @@
+        maxSeekHole?: number;
+        seekHoleNudgeDuration?: number;
+        nudgeOffset?: number;
+        nudgeMaxRetry?: number;
+        maxFragLookUpTolerance?: number;
+        liveSyncDurationCount?: number;
+        liveMaxLatencyDurationCount?: number;
+        enableWorker?: boolean;
+        enableSoftwareAES?: boolean;
+        manifestLoadingTimeOut?: number;
+        manifestLoadingMaxRetry?: number;
+        manifestLoadingRetryDelay?: number;
+        manifestLoadingMaxRetryTimeout ?: number;
+        levelLoadingTimeOut?: number;
+        levelLoadingMaxRetry?: number;
+        levelLoadingRetryDelay?: number;
+        levelLoadingMaxRetryTimeout?: number;
+        fragLoadingTimeOut?: number;
+        fragLoadingMaxRetry?: number;
+        fragLoadingRetryDelay?: number;
+        fragLoadingMaxRetryTimeout?: number;
+        fragLoadingLoopThreshold?: number;
+        startFragPrefech?: boolean;
+        appendErrorMaxRetry?: number;
+        loader?: Loader;
+        fLoader?: Loader;
+        pLoader?: Loader;
+        xhrSetup?: (xhr?: XMLHttpRequest, url?: string) => void;
+        fetchSetup?: (context?: LoaderContext, params?: any) => void;
+        abrController?: any;
+        timelineController?: any;
+        enableCEA708Captions?: boolean;
+        stretchShortVideoTrack?: boolean;
+        forceKeyFrameOnDiscontinuity?: boolean;
+        abrEwmaFastLive?: number;
+        abrEwmaSlowLive?: number;
+        abrEwmaFastVoD?: number;
+        abrEwmaSlowVoD?: number;
+        abrEwmaDefaultEstimate?: number;
+        abrBandWidthFactor?: number;
+        abrBandWidthUpFactor?: number;
+    }
+
+    export interface ErrorEvent {
+        type: ErrorType;
+        details: ErrorDetails;
+        fatal: boolean;
+    }
+
+    export interface QualityLevel {
+        attrs: any; // todo
+        details?: any; // todo
+
+        audioCodec: string;
+        videoCodec: string;
+        bitrate: number;
+        url: string[];
+        urlId: number;
+        name: string;
+        width: number;
+        height: number;
+    }
+
+    export interface Instance {
+        /**
+         * Calling this method will:
+         *  - bind videoElement and hls instance;
+         *  - create MediaSource and set it as video source
+         *  - once MediaSource object is successfully created;
+         *    MEDIA_ATTACHED event will be fired.
+         */
+        attachMedia(element: HTMLVideoElement): void;
+
+        /**
+         * Calling this method will:
+         *  - unbind VideoElement from hls instance;
+         *  - signal the end of the stream on MediaSource
+         *  - reset video source (video.src = '')
+         */
+        detachMedia(): void;
+
+        /**
+         * Should be called to free used resources and destroy hls context.
+         */
+        destroy(): void;
+
+        /**
+         * Starts loading video data from the manifest URL.
+         */
+        loadSource(url: string): void;
```

Comment:
I notice that `hls.startLoad(startPosition=-1)` and `hls.stopLoad()` is missing.
https://github.com/dailymotion/hls.js/blob/master/doc/API.md#hlsstopload
https://github.com/dailymotion/hls.js/blob/master/doc/API.md#hlsstartloadstartposition-1